alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultConfigurationAccessor.java

```diff
@@ -167,15 +167,22 @@ public void deleteJob(UUID jobId) throws AlertDatabaseConstraintException {
         if (StringUtils.isBlank(providerConfigName)) {
             throw new AlertDatabaseConstraintException("The provider configuration name cannot be null");
         }
-        Long fieldId = definedFieldRepository.findFirstByKey(ProviderGlobalUIConfig.KEY_PROVIDER_CONFIG_NAME)
+        Long fieldId = definedFieldRepository.findFirstByKey(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME)
                            .map(DefinedFieldEntity::getId)
-                           .orElseThrow(() -> new AlertDatabaseConstraintException(String.format("The key '%s' is not registered in the database", ProviderGlobalUIConfig.KEY_PROVIDER_CONFIG_NAME)));
-        Optional<Long> optionalProviderConfigId = fieldValueRepository.findAllByFieldIdAndValue(fieldId, providerConfigName)
-                                                      .stream()
-                                                      .map(FieldValueEntity::getConfigId)
-                                                      .findFirst();
-        if (optionalProviderConfigId.isPresent()) {
-            return getConfigurationById(optionalProviderConfigId.get());
+                           .orElseThrow(() -> new AlertDatabaseConstraintException(String.format("The key '%s' is not registered in the database", ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME)));
+        List<Long> providerConfigIds = fieldValueRepository.findAllByFieldIdAndValue(fieldId, providerConfigName)
+                                           .stream()
+                                           .map(FieldValueEntity::getConfigId)
+                                           .collect(Collectors.toList());
+        if (!providerConfigIds.isEmpty()) {
+            for (Long configId : providerConfigIds) {
+                Optional<ConfigurationModel> configurationModel = getConfigurationById(configId);
+                Optional<ConfigurationModel> globalModel = configurationModel
+                                                               .filter(model -> model.getDescriptorContext() == ConfigContextEnum.GLOBAL);
+                if (globalModel.isPresent()) {
+                    return globalModel;
+                }
```

Comment:
These lines can all be collapsed into something similar to
`getConfigurationById(configId).filter(model -> model.getDescriptorContext() == ConfigContextEnum.GLOBAL);`
Although exception handling could cause problems here.